### PR TITLE
[kitchen] Add SUSE 11 kitchen tests

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Adds kitchen tests for SUSE 11 with sysvinit.
Fixes problems with said sysvinit scripts.

### Motivation

Check that the sysvinit scripts we created do work correctly.

### Additional Notes

A5 upgrade tests are disabled for now because the A5 install-script does not work on SUSE 11.
A6 upgrade tests are disabled for now as we don't have a stable A6 version supporting SUSE 11 yet.

### CODE REVIEW

File: cmd/agent/install_script.sh [hunk 1]
```diff
@@ -208,12 +207,24 @@ elif [ $OS = "SUSE" ]; then
...
-  $sudo_cmd rpm --import https://yum.${repo_url}/DATADOG_RPM_KEY.public
-  $sudo_cmd rpm --import https://yum.${repo_url}/DATADOG_RPM_KEY_E09422B3.public
+  if [ "$SUSE11" == "yes" ]; then
+    $sudo_cmd curl -o /tmp/DATADOG_RPM_KEY.public https://yum.${repo_url}/DATADOG_RPM_KEY.public
+    $sudo_cmd rpm --import /tmp/DATADOG_RPM_KEY.public
+    $sudo_cmd curl -o /tmp/DATADOG_RPM_KEY_E09422B3.public https://yum.${repo_url}/DATADOG_RPM_KEY_E09422B3.public
+    $sudo_cmd rpm --import /tmp/DATADOG_RPM_KEY_E09422B3.public
+  else
+    $sudo_cmd rpm --import https://yum.${repo_url}/DATADOG_RPM_KEY.public
+    $sudo_cmd rpm --import https://yum.${repo_url}/DATADOG_RPM_KEY_E09422B3.public
```
Comment (albertvaka) [inline cmd/agent/install_script.sh]: We could import only the new key when installing A7. This would serve to check we have signed it with the correct key. We can do this to a different PR

File: omnibus/package-scripts/agent/posttrans [hunk 2]
```diff
@@ -41,10 +41,10 @@ if command -v systemctl >/dev/null 2>&1; then
 elif command -v initctl >/dev/null 2>&1; then
     # start/stop policy is already defined in the upstart job file
     :
-elif command -v update-rc.d >/dev/null 2>&1; then
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/posttrans]: Shouldn't we keep this `elif` block after the new `elif` block you added? Can we know it's never used?

File: omnibus/package-scripts/agent/prerm [hunk 3]
```diff
@@ -29,10 +35,9 @@ stop_agent()
         initctl stop $SERVICE_NAME-sysprobe || true
         initctl stop $SERVICE_NAME-trace || true
         initctl stop $SERVICE_NAME || true
-    elif [ "$DISTRIBUTION_FAMILY" = "Debian" ]; then
+    elif [ "$DISTRIBUTION_FAMILY" = "Debian" ] || [ "$SUSE_SYSVINIT_SUPPORT"="yes" ]; then
         if command -v service >/dev/null 2>&1; then
             service $SERVICE_NAME-process stop || true
-            service $SERVICE_NAME-sysprobe stop || true
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/prerm]: In Debian we might want to keep this line still.

File: omnibus/package-scripts/agent/prerm [hunk 4]
```diff
@@ -59,12 +64,15 @@ deregister_agent()
     elif [ "$DISTRIBUTION_FAMILY" = "Debian" ]; then
         if command -v update-rc.d >/dev/null 2>&1; then
             update-rc.d -f $SERVICE_NAME-process remove || true
-            update-rc.d -f $SERVICE_NAME-sysprobe remove || true
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/prerm]: Idem.

File: omnibus/package-scripts/agent/prerm [hunk 5]
```diff
@@ -59,12 +64,15 @@ deregister_agent()
...
             update-rc.d -f $SERVICE_NAME-trace remove || true
             update-rc.d -f $SERVICE_NAME remove || true
         else
             echo "[ WARNING ]\tCannot detect a supported init system. The datadog-agent package only provides service files for systemd, upstart and sysvinit."
         fi
+    elif [ "$SUSE_SYSVINIT_SUPPORT"="yes" ]; then
```
Comment (albertvaka) [inline omnibus/package-scripts/agent/prerm]: Not sure if this works without spaces between the operator and operands.

File: omnibus/package-scripts/agent/prerm (hunk 3 above)
Comment (KSerrania) [reply omnibus/package-scripts/agent/prerm]: The sysprobe SysVInit script for debian was never shipped with the Agent, see #4437

File: omnibus/package-scripts/agent/posttrans (hunk 2 above)
Comment (KSerrania) [reply omnibus/package-scripts/agent/posttrans]: I added it specifically for SUSE 11 (this block didn't exist before). Since I realised it doesn't work on SUSE 11 either by default, I removed it. We could keep it within the `$SUSE_SYSVINIT_SUPPORT` elif block (in case someone does use `update-rc.d` on SUSE 11), in which case I'll have to re-add it back to the prerm script as well.

File: omnibus/package-scripts/agent/prerm (hunk 5 above)
Comment (KSerrania) [reply omnibus/package-scripts/agent/prerm]: Indeed, it only worked in the tests we do because this always returns true.

File: omnibus/package-scripts/agent/prerm (hunk 3 above)
Comment (albertvaka) [reply omnibus/package-scripts/agent/prerm]: Does this only apply to SysVinit, though? Or also to upstart?

Comment (KSerrania) [reply omnibus/package-scripts/agent/prerm]: I don't think so, as Upstart-managed debian / ubuntu systems should have been caught in the previous `initctl` block (since `initctl` is the [init control tool for Upstart](http://manpages.ubuntu.com/manpages/trusty/man8/initctl.8.html)). However, there _might_ be weird cases where a user could have upstart without `initctl`, using only `service`, or `start` / `stop` directly. I can leave this line here for now if we want to make sure we don't break any use case.

Comment (KSerrania) [reply omnibus/package-scripts/agent/prerm]: I re-added the line, with a comment

**Review by albertvaka — APPROVED** [review]
